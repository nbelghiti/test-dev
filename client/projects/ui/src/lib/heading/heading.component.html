<div
	class="c-heading {{ classes }}"
	[ngClass]="{'c-heading--inline': inline}"
>
	<ng-container [ngSwitch]="level">
		<h1 *ngSwitchCase="HeadingLevel.heading1">{{title}}</h1>
		<h2 *ngSwitchCase="HeadingLevel.heading2">{{title}}</h2>
		<h3 *ngSwitchCase="HeadingLevel.heading3">{{title}}</h3>
	</ng-container>

	<div class="c-heading__meta">
		<i
			*ngIf="loading; else withContent"
			class="fas fa-{{IconName.spinner}} fa-spin c-heading__loader"
		>
		</i>

		<ng-template #withContent>
			<ng-content></ng-content>
		</ng-template>
	</div>
</div>
