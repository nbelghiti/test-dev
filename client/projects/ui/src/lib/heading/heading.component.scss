// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Heading
// =============================================================================
.c-heading {
	display: inline-flex;
	flex-direction: column;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-heading__loader {
	color: $primary-color;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-heading--inline {
	flex-direction: row;
	align-items: baseline;

	.c-heading__meta {
		margin-top: 0;
		margin-left: 1rem;
	}
}
