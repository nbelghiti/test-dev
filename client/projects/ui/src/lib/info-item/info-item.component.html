<div class="c-info-item">
	<ui-icon [name]="icon"></ui-icon>

	<span>
		{{ text || '-' }}
	</span>
</div>
