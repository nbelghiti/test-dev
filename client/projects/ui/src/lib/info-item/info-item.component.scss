// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Settings
// =============================================================================

// =============================================================================
// :: Info Item
// =============================================================================
.c-info-item {
	display: flex;
	align-items: center;

	font-family: $font-family-secondary;
	font-size: $font-size-1x-small;
	line-height: $line-height-base;

	color: $text-color-base;

	.c-icon {
		width: $font-size-1x-small * $line-height-base;
		margin-right: 1rem;
	}
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-info-item--uppercase {
	text-transform: uppercase;
}

// .c-list--ordered {
// 	counter-reset: list;

// 	.c-list__item {

// 		&::before {
// 			content: counter(list) ".";

// 			margin-right: 0.7rem;

// 			counter-increment: list;
// 		}
// 	}
// }

