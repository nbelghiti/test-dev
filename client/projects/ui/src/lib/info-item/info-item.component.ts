import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

@Component({
	selector: 'ui-info-item',
	templateUrl: './info-item.component.html',
	styleUrls: ['./info-item.component.scss']
})

export class InfoItemComponent {
	@Input() public icon: IconName;
	@Input() public text: string;
}
