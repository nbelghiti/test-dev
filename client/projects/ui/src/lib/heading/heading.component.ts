import { Component, Input } from '@angular/core';

import { IconName } from '../icon/icon.types';

import { HeadingLevel } from './heading.types';

@Component({
	selector: 'ui-heading',
	templateUrl: './heading.component.html',
	styleUrls: ['./heading.component.scss']
})

export class HeadingComponent {
	@Input() public title: string;
	@Input() public classes: string = '';
	@Input() public inline: boolean = false;
	@Input() public loading: boolean = false;
	@Input() public level: HeadingLevel = HeadingLevel.heading1;

	public HeadingLevel: typeof HeadingLevel = HeadingLevel;
	public IconName: typeof IconName = IconName;
}
