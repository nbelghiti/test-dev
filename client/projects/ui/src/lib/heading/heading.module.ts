import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { HeadingComponent } from './heading.component';

@NgModule({
	imports: [
		CommonModule,
	],
	declarations: [
		HeadingComponent,
	],
	exports: [
		HeadingComponent,
	],
})
export class HeadingModule {}
