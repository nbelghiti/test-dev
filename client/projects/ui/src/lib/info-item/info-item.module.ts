import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { IconModule } from '../icon/icon.module';

import { InfoItemComponent } from './info-item.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
	],
	declarations: [
		InfoItemComponent,
	],
	exports: [
		InfoItemComponent,
	],
})
export class InfoItemModule {}
