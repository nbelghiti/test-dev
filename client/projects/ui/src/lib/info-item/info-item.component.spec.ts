import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';

import { HeadingComponent } from '../heading/heading.component';
import { IconComponent } from '../icon/icon.component';

import { InfoItemComponent } from './info-item.component';

describe('InfoItemComponent', () => {
	let component: InfoItemComponent;
	let fixture: ComponentFixture<InfoItemComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
			],
			declarations: [
				InfoItemComponent,
				IconComponent,
				HeadingComponent
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(InfoItemComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

