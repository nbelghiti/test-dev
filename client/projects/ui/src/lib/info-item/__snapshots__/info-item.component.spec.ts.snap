// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoItemComponent Should render correctly 1`] = `
<ui-info-item>
  <div
    class="c-info-item"
  >
    <ui-icon
      class="c-icon"
    >
      <i
        class="c-icon__glyph   fas fa-"
      />
    </ui-icon>
    <span>
       - 
    </span>
  </div>
</ui-info-item>
`;
